Master branch fails tests using Python 3.6.3

In response to the test failures in issue #1620, I ran the tests using Python 3.6.3 and received the same errors as the [travis run](https://travis-ci.org/open-source-economics/Tax-Calculator/jobs/298126321).  Further, I ran the tests on the master branch and received the same errors.  

```
================================================================================================================================== FAILURES ==================================================================================================================================
_____________________________________________________________________________________________________________________________ test_create_tables _____________________________________________________________________________________________________________________________

cps_subsample =         age_head  age_spouse      e00200p      e00900p      e02100p  \
445636        41           0   922,485.99      ...2  
52701       0    0        5  
66990       0    0       16  
428804      0    0        9  

[4565 rows x 54 columns]

    def test_create_tables(cps_subsample):
        # pylint: disable=too-many-statements
        # create a current-law Policy object and Calculator object calc1
        rec = Records.cps_constructor(data=cps_subsample)
        pol = Policy()
        calc1 = Calculator(policy=pol, records=rec)
        calc1.calc_all()
        # create a policy-reform Policy object and Calculator object calc2
        reform = {2013: {'_II_rt1': [0.15]}}
        pol.implement_reform(reform)
        calc2 = Calculator(policy=pol, records=rec)
        calc2.calc_all()
    
        # test creating various difference tables
    
        diff = create_difference_table(calc1.records, calc2.records,
                                       groupby='large_income_bins',
                                       income_measure='expanded_income',
                                       tax_to_diff='combined')
        assert isinstance(diff, pd.DataFrame)
        expected = [0.00,
                    0.01,
                    0.41,
                    0.84,
                    0.92,
                    1.10,
                    1.15,
                    1.04,
                    0.78,
                    0.27,
                    np.nan]
        assert np.allclose(diff['perc_aftertax'], expected,
                           atol=0.005, rtol=0.0, equal_nan=True)
    
        diff = create_difference_table(calc1.records, calc2.records,
                                       groupby='webapp_income_bins',
                                       income_measure='expanded_income',
                                       tax_to_diff='iitax')
        assert isinstance(diff, pd.DataFrame)
        expected = [0.00,
                    0.01,
                    0.41,
                    0.84,
                    0.92,
                    1.10,
                    1.15,
                    1.04,
                    0.78,
                    0.30,
                    0.08,
                    0.07,
                    np.nan]
        assert np.allclose(diff['perc_aftertax'], expected,
                           atol=0.005, rtol=0.0, equal_nan=True)
    
        diff = create_difference_table(calc1.records, calc2.records,
                                       groupby='small_income_bins',
                                       income_measure='expanded_income',
                                       tax_to_diff='iitax')
        assert isinstance(diff, pd.DataFrame)
        expected = [0.00,
                    0.01,
                    0.02,
                    0.16,
                    0.64,
                    0.82,
                    0.87,
                    0.92,
                    1.10,
                    1.15,
                    1.04,
                    0.78,
                    0.30,
                    0.08,
                    0.09,
                    0.07,
                    0.05,
                    0.02,
                    0.00,
                    np.nan]
>       assert np.allclose(diff['perc_aftertax'], expected,
                           atol=0.005, rtol=0.0, equal_nan=True)
E       assert False
E        +  where False = <function allclose at 0x7f98cb85a1e0>(0            0.00\n1            0.01\n2            0.02\n3            0.16\n4            0.64\n5            0.82\n6         ...       0.07\n16           0.05\n17           0.02\n18            nan\nsums          nan\nName: perc_aftertax, dtype: float64, [0.0, 0.01, 0.02, 0.16, 0.64, 0.82, ...], atol=0.005, rtol=0.0, equal_nan=True)
E        +    where <function allclose at 0x7f98cb85a1e0> = np.allclose

tests/test_utils.py:143: AssertionError
---------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
________________________________________________________________________________________________________________________ test_weighted_count_lt_zero _________________________________________________________________________________________________________________________

    def test_weighted_count_lt_zero():
        df1 = pd.DataFrame(data=DATA, columns=['tax_diff', 's006', 'label'])
        grped = df1.groupby('label')
        diffs = grped.apply(weighted_count_lt_zero, 'tax_diff')
        exp = pd.Series(data=[4, 0], index=['a', 'b'])
        exp.index.name = 'label'
>       pd.util.testing.assert_series_equal(exp, diffs)

tests/test_utils.py:489: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1238: in assert_series_equal
    assert_attr_equal('dtype', left, right)
../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1008: in assert_attr_equal
    raise_assert_detail(obj, msg, left_attr, right_attr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'Attributes', message = 'Attribute "dtype" are different', left = dtype('int64'), right = dtype('float64'), diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{obj} are different
    
    {message}
    [left]:  {left}
    [right]: {right}""".format(obj=obj, message=message, left=left, right=right)
    
        if diff is not None:
            msg += "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  int64
E       [right]: float64

../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1089: AssertionError
___________________________________________________________________________________________________________________________ test_weighted_perc_cut ___________________________________________________________________________________________________________________________

    def test_weighted_perc_cut():
        dfx = pd.DataFrame(data=DATA, columns=['tax_diff', 's006', 'label'])
        grouped = dfx.groupby('label')
        diffs = grouped.apply(weighted_perc_cut, 'tax_diff')
        exp = pd.Series(data=[4. / 12., 0.0], index=['a', 'b'])
        exp.index.name = 'label'
>       pd.util.testing.assert_series_equal(exp, diffs)

tests/test_utils.py:574: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1272: in assert_series_equal
    obj='{obj}'.format(obj=obj))
pandas/_libs/testing.pyx:59: in pandas._libs.testing.assert_almost_equal
    ???
pandas/_libs/testing.pyx:173: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'Series', message = 'Series values are different (50.0 %)', left = '[0.333333333333, 0.0]', right = '[0.333333333306, nan]', diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{obj} are different
    
    {message}
    [left]:  {left}
    [right]: {right}""".format(obj=obj, message=message, left=left, right=right)
    
        if diff is not None:
            msg += "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: Series are different
E       
E       Series values are different (50.0 %)
E       [left]:  [0.333333333333, 0.0]
E       [right]: [0.333333333306, nan]

../../miniconda3/envs/taxcalc-dev/lib/python3.6/site-packages/pandas/util/testing.py:1089: AssertionError
__________________________________________________________________________________________________________________________ test_diff_table_sum_row ___________________________________________________________________________________________________________________________

cps_subsample =         age_head  age_spouse      e00200p      e00900p      e02100p  \
445636        41           0   922,485.99      ...2  
52701       0    0        5  
66990       0    0       16  
428804      0    0        9  

[4565 rows x 54 columns]

    def test_diff_table_sum_row(cps_subsample):
        rec = Records.cps_constructor(data=cps_subsample)
        # create a current-law Policy object and Calculator calc1
        pol = Policy()
        calc1 = Calculator(policy=pol, records=rec)
        calc1.calc_all()
        # create a policy-reform Policy object and Calculator calc2
        reform = {2013: {'_II_rt4': [0.56]}}
        pol.implement_reform(reform)
        calc2 = Calculator(policy=pol, records=rec)
        calc2.calc_all()
        # create two difference tables and compare their content
        tdiff1 = create_difference_table(calc1.records, calc2.records,
                                         groupby='small_income_bins',
                                         income_measure='expanded_income',
                                         tax_to_diff='iitax')
        tdiff2 = create_difference_table(calc1.records, calc2.records,
                                         groupby='large_income_bins',
                                         income_measure='expanded_income',
                                         tax_to_diff='iitax')
        non_digit_cols = ['mean', 'perc_inc', 'perc_cut', 'share_of_change',
                          'perc_aftertax', 'pc_aftertaxinc']
        digit_cols = [c for c in list(tdiff1) if c not in non_digit_cols]
>       assert np.allclose(tdiff1[digit_cols][-1:],
                           tdiff2[digit_cols][-1:])
E       assert False
E        +  where False = <function allclose at 0x7f98cb85a1e0>(            count    tax_cut    tax_inc     tot_change\nsums 1,641,299.36        nan  65,138.69 806,980,455.93,             count    tax_cut    tax_inc     tot_change\nsums 1,641,299.36        nan  65,138.69 806,980,455.93)
E        +    where <function allclose at 0x7f98cb85a1e0> = np.allclose

tests/test_utils.py:716: AssertionError
---------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
============================================================================================================================ 68 tests deselected =============================================================================================================================

```

--- a/taxcalc/tests/test_utils.py
+++ b/taxcalc/tests/test_utils.py
@@ -91,7 +91,7 @@ def test_create_tables(cps_subsample):
                 0.78,
                 0.27,
                 np.nan]
-    assert np.allclose(diff['perc_aftertax'], expected,
+    assert np.allclose(diff['perc_aftertax'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)
 
     diff = create_difference_table(calc1.records, calc2.records,
@@ -112,7 +112,7 @@ def test_create_tables(cps_subsample):
                 0.08,
                 0.07,
                 np.nan]
-    assert np.allclose(diff['perc_aftertax'], expected,
+    assert np.allclose(diff['perc_aftertax'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)
 
     diff = create_difference_table(calc1.records, calc2.records,
@@ -138,9 +138,10 @@ def test_create_tables(cps_subsample):
                 0.07,
                 0.05,
                 0.02,
-                0.00,
+                np.nan,
                 np.nan]
-    assert np.allclose(diff['perc_aftertax'], expected,
+    print('here', diff['perc_aftertax'], diff['perc_aftertax'] - expected)
+    assert np.allclose(diff['perc_aftertax'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)
 
     diff = create_difference_table(calc1.records, calc2.records,
@@ -162,7 +163,7 @@ def test_create_tables(cps_subsample):
                 66560891,
                 39571078,
                 5282069]
-    assert np.allclose(diff['tot_change'], expected,
+    assert np.allclose(diff['tot_change'].values, expected,
                        atol=0.5, rtol=0.0)
     expected = [0.00,
                 0.05,
@@ -178,7 +179,7 @@ def test_create_tables(cps_subsample):
                 12.38,
                 7.36,
                 0.98]
-    assert np.allclose(diff['share_of_change'], expected,
+    assert np.allclose(diff['share_of_change'].values, expected,
                        atol=0.005, rtol=0.0)
     expected = [0.00,
                 0.02,
@@ -194,7 +195,7 @@ def test_create_tables(cps_subsample):
                 0.70,
                 0.37,
                 0.06]
-    assert np.allclose(diff['perc_aftertax'], expected,
+    assert np.allclose(diff['perc_aftertax'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)
     expected = [-0.00,
                 -0.02,
@@ -210,7 +211,7 @@ def test_create_tables(cps_subsample):
                 -0.70,
                 -0.37,
                 -0.06]
-    assert np.allclose(diff['pc_aftertaxinc'], expected,
+    assert np.allclose(diff['pc_aftertaxinc'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)
 
     with pytest.raises(ValueError):
@@ -241,7 +242,7 @@ def test_create_tables(cps_subsample):
                 4171055704,
                 2751003155,
                 709210048]
-    assert np.allclose(dist['iitax'], expected,
+    assert np.allclose(dist['iitax'].values, expected,
                        atol=0.5, rtol=0.0)
     expected = [1202,
                 1688,

--- a/taxcalc/tests/test_utils.py
+++ b/taxcalc/tests/test_utils.py
@@ -140,7 +140,6 @@ def test_create_tables(cps_subsample):
                 0.02,
                 np.nan,
                 np.nan]
-    print('here', diff['perc_aftertax'], diff['perc_aftertax'] - expected)
     assert np.allclose(diff['perc_aftertax'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)
 

--- a/taxcalc/tests/test_utils.py
+++ b/taxcalc/tests/test_utils.py
@@ -138,7 +138,7 @@ def test_create_tables(cps_subsample):
                 0.07,
                 0.05,
                 0.02,
-                np.nan,
+                0.0,
                 np.nan]
     assert np.allclose(diff['perc_aftertax'].values, expected,
                        atol=0.005, rtol=0.0, equal_nan=True)

--- a/taxcalc/utilsprvt.py
+++ b/taxcalc/utilsprvt.py
@@ -5,6 +5,8 @@ PRIVATE utility functions for Tax-Calculator PUBLIC utility functions.
 # pep8 --ignore=E402 utilsprvt.py
 # pylint --disable=locally-disabled utilsprvt.py
 
+import numpy as np
+
 
 EPSILON = 1e-9
 
@@ -12,13 +14,18 @@ EPSILON = 1e-9
 def weighted_count_lt_zero(pdf, col_name, tolerance=-0.001):
     """
     Return weighted count of negative Pandas DateFrame col_name items.
+    If condition is not met by any items, the result of applying sum to an
+    empty dataframe is NaN.  This is undesirable and 0 is returned instead.
     """
-    return pdf[pdf[col_name] < tolerance]['s006'].sum()
+    res = pdf[pdf[col_name] < tolerance]['s006'].sum()
+    return res if not np.isnan(res) else 0
 
 
 def weighted_count_gt_zero(pdf, col_name, tolerance=0.001):
     """
     Return weighted count of positive Pandas DateFrame col_name items.
+    If condition is not met by any items, the result of applying sum to an
+    empty dataframe is NaN.  This is undesirable and 0 is returned instead.
     """
     return pdf[pdf[col_name] > tolerance]['s006'].sum()
 

--- a/taxcalc/utilsprvt.py
+++ b/taxcalc/utilsprvt.py
@@ -41,8 +41,11 @@ def weighted_mean(pdf, col_name):
     """
     Return weighted mean of Pandas DataFrame col_name items.
     """
-    return ((pdf[col_name] * pdf['s006']).sum() /
-            (pdf['s006'].sum() + EPSILON))
+    if len(pdf) > 0:
+        return ((pdf[col_name] * pdf['s006']).sum() /
+                (pdf['s006'].sum() + EPSILON))
+    else:
+        return 0
 
 
 def wage_weighted(pdf, col_name):

--- a/taxcalc/utilsprvt.py
+++ b/taxcalc/utilsprvt.py
@@ -27,7 +27,8 @@ def weighted_count_gt_zero(pdf, col_name, tolerance=0.001):
     If condition is not met by any items, the result of applying sum to an
     empty dataframe is NaN.  This is undesirable and 0 is returned instead.
     """
-    return pdf[pdf[col_name] > tolerance]['s006'].sum()
+    res = pdf[pdf[col_name] > tolerance]['s006'].sum()
+    return res if not np.isnan(res) else 0
 
 
 def weighted_count(pdf):

